client/etcd/client.go

```diff
@@ -311,7 +309,7 @@ func validateTopLevelNamespace(namespace string) error {
 }
 
 func (c *csclient) sanitizeOptions(opts kv.Options) (kv.Options, error) {
-	if opts.Logger() == nil {
+	if opts.Logger() == nil || reflect.DeepEqual(opts.Logger(), log.NullLogger) {
```

Comment:
We can probably just do `opts.Logger() == log.NullLogger` here?

Reply:
We can't the NullLogger is not comparable, I guess because of the slice of fields in it.

Reply:
Sure, but `log.NullLogger` is a singleton and of type `Logger`, which is an interface, we should be able to compare interface pointers no?

Reply:
Are you saying we should compare the address of the struct under the log.Logger interface? but I don't see how can we we get that without using reflect or unsafe pointer? Please feel free to commandeer this pr if there is an easy way to do it

Reply:
What I meant was `NullLogger` is a singleton and of `Logger` type, which is an interface. Therefore under the hood, `NullLogger` contains a pointer to the slice fields in the null logger on the heap. As such, checking `opts.Logger() == log.NullLogger` should just work. FWIW I just tried it and it builds locally

Reply:
Ah I see, didn't realize the `NullLogger` wasn't a pointer type by itself, which would cause a panic during runtime. Put up a m3x PR here to make `NullLogger` a pointer type: https://github.com/m3db/m3x/pull/108 and updated this PR to use direct logger comparison without reflection.